src/mattack_actors.cpp

```diff
@@ -184,6 +188,25 @@ bool mon_spellcasting_actor::call( monster &mon ) const
         return false;
     }
 
+    std::string target_name = "";
+    const monster *target_monster = g->critter_at<monster>( target );
```

Comment:
```suggestion
    if( const Creature *const target_critter = g->critter_at( target ) ) {
         target_name = target_critter->disp_name();
     }
```
In all of the below branches, you call `disp_name`, but that exists within the `Creature` class already.

Reply:
Sorry I don't really understand what you mean?

Reply:
`disp_name` is a virtual function. You can call it on a `Creature` object, and it will call `monster::disp_name` or `npc::disp_name` or `disp_name` of whatever else class inherits from `Creature`. You don't need to replicate that behavior by checking which type of creature is there. See https://en.cppreference.com/w/cpp/language/virtual

Reply:
I see. I did in fact try this first and were unable to. Iirc the problem was that I were not able to say `creature entity = ..` . I do not remember the exact error, but something like "can't instantiate abstract class".

Reply:
i see now. I was trying to do `g->critter_at<Creature>`. thanks